lib/yaml_permitted_classes.rb

```diff
@@ -0,0 +1,25 @@
+class YamlPermittedClasses
+  DEFAULT_PERMITTED_CLASSES = [Object, Range, Regexp, Symbol, Date, Time, DateTime, ActiveSupport::Duration, ActiveSupport::HashWithIndifferentAccess, ActiveSupport::TimeWithZone, ActiveSupport::TimeZone]
+  def self.app_yaml_permitted_classes
+    @app_yaml_permitted_classes ||= DEFAULT_PERMITTED_CLASSES + [MiqExpression, MiqReport, Ruport::Data::Table, Ruport::Data::Record, User, ConfigurationScript, ContainerImage, ContainerTemplate, OpenStruct, OrchestrationTemplate, ManageIQ::Providers::Vmware::InfraManager, ManageIQ::Providers::InfraManager::Vm, VimHash, VimString, VimArray, ActiveModel::Type::String, ActiveModel::Attribute.const_get(:FromDatabase), ActiveModel::Attribute.const_get(:FromUser)]
```

Comment:
If they are test only, then I think that plugin's RSpec config should be setting it instead of it being set globally.

Reply:
Yeah we dropped the serialized VimTypes in yaml columns a while ago and shouldn't be needed

Reply:
> If they are test only, then I think that plugin's RSpec config should be setting it instead of it being set globally.

This is proactively moving us to using safe_load on yaml columns so I would need confidence we're not serializing these types in columns in addition to the tests/other code doing yaml load.  I'll look again but that was the thought when I added them a whiel ago.

Reply:
If we need to load `ManageIQ::Providers::Vmware::InfraManager` we probably should have `ExtManagementSystem` instead of the child class since I'm guessing that was just the one being used in specs but it could be others.

Reply:
In case your curious about `FromDatabase` and friends, that's from serializing unpersisted objects...

From the commit:
```
We're adding active model classes due to:

 puts MiqReport.first.to_yaml
--- !ruby/object:MiqReport
concise_attributes:
- !ruby/object:ActiveModel::Attribute::FromDatabase
  name: id
  value_before_type_cast: 1
- !ruby/object:ActiveModel::Attribute::FromDatabase
  name: name
...


Could be related to https://github.com/rails/rails/issues #25145
```